### PR TITLE
Warns when npm or bower dependencies aren't installed

### LINKED ISSUE #3961: Inside of a freshly cloned app: `Cannot find module 'ember-cli/lib/broccoli/ember-app'`
When I clone an 0.2.3 app and try to run `ember serve`, I get an error:

```
ejb-mba:testapp ebryn$ ember serve
Future versions of Ember CLI will not support v0.10.26. Please update to Node 0.12 or io.js.
version: 0.2.3
Could not find watchman, falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/#watchman for more info.
Cannot find module 'ember-cli/lib/broccoli/ember-app'
Error: Cannot find module 'ember-cli/lib/broccoli/ember-app'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/ebryn/Dev/junk/testapp/Brocfile.js:3:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

I should be told that I need to `npm install && bower install`.

### PR DESCRIPTION
**Pull request by marcioj**
Closes #3961

### CODE REVIEW

**Comment by stefanpenner** [comment]
lvgtm thank you sir

File: tests/unit/models/installation-checker-test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+var expect              = require('chai').expect;
+var InstallationChecker = require('../../../lib/models/installation-checker');
+var path                = require('path');
+
+describe.only('Installation Checker', function() {
```
Comment (marcioj) [inline tests/unit/models/installation-checker-test.js]: Oh, I forgot this `only` here. I will send a fix now.

Comment (stefanpenner) [inline tests/unit/models/installation-checker-test.js]: @marcioj we should add a test that validates no only's are submitted...

Comment (marcioj) [inline tests/unit/models/installation-checker-test.js]: @stefanpenner :+1:. I will try something tonight

Comment (stefanpenner) [inline tests/unit/models/installation-checker-test.js]: :+1:

**Pull request merged**